piptools/_compat/pip_compat.py

```diff
@@ -169,11 +174,26 @@ def _normalize_comes_from_location(original_comes_from: str, /) -> str:
     # split on the space
     prefix, space_sep, suffix = original_comes_from.partition(" ")
 
+    # if the value part is a remote URI for pip, return the original
+    if _is_remote_pip_uri(suffix):
+        return original_comes_from
+
     # convert to a posix-style path
     suffix = pathlib.Path(suffix).as_posix()
     return f"{prefix}{space_sep}{suffix}"
 
 
+def _is_remote_pip_uri(value: str) -> bool:
+    """
+    Test a string to see if it is a URI treated as a remote file in ``pip``.
+    Specifically this means that it's a 'file', 'http', or 'https' URI.
+
+    The test is performed by trying a URL parse and reading the scheme.
+    """
+    scheme = urllib.parse.urlsplit(value).scheme
+    return scheme in ("http", "https", "file")
```

Comment:
A set reads slightly better semantically.
```suggestion
    return scheme in {"file", "http", "https"}
```